cub/test/catch2_test_block_radix_sort.cuh

```diff
@@ -366,22 +366,63 @@ get_striped_keys(const thrust::host_vector<KeyT> &h_keys,
   thrust::host_vector<KeyT> h_striped_keys(h_keys);
   KeyT *h_striped_keys_data = thrust::raw_pointer_cast(h_striped_keys.data());
 
-  if ((begin_bit > 0) || (end_bit < static_cast<int>(sizeof(KeyT) * 8)))
+  using traits_t                    = cub::Traits<KeyT>;
+  using bit_ordered_t               = typename traits_t::UnsignedBits;
+  const bit_ordered_t negative_zero = bit_ordered_t{1} << bit_ordered_t{sizeof(bit_ordered_t) * 8 - 1};
+
+  const int num_bits = end_bit - begin_bit;
+
+  for (std::size_t i = 0; i < h_keys.size(); i++)
   {
-    const int num_bits = end_bit - begin_bit;
+    bit_ordered_t key = reinterpret_cast<const bit_ordered_t&>(h_keys[i]);
+
+    if (traits_t::CATEGORY == cub::FLOATING_POINT && key == negative_zero)
+    {
+      key = 0;
+    }
+
+    key = traits_t::TwiddleIn(key);
 
-    for (std::size_t i = 0; i < h_keys.size(); i++)
+    if ((begin_bit > 0) || (end_bit < static_cast<int>(sizeof(KeyT) * 8)))
     {
       unsigned long long base = 0;
-      memcpy(&base, h_striped_keys_data + i, sizeof(KeyT));
+      memcpy(&base, &key, sizeof(bit_ordered_t));
```

Comment:
fyi: `SafeBitCast<T>(in)` from `test_util.cuh` does this with type deduction and static size checks.

Reply:
New tests can't include `test_util.cuh`. Once we get rid of old tests, we can pull `SafeBitCast<T>(in)` into c2h and use it here.